### PR TITLE
Add produces option for /forgot

### LINKED ISSUE #326: Implement produces option for /forgot
Implement the produces option for `/forgot` according to the [framework spec](https://github.com/stormpath/stormpath-framework-spec/blob/master/password-reset.md#request-handling):

>##### GET requests with Accept: text/html:

>If stormpath.web.spa.enabled is false, render a form which allows the user to request a password reset by entering their email address. If the request contains the query ?status=invalid_sptoken, a message should be shown above the form:

>The password reset link you tried to use is no longer valid. Please request a new link from the form below. If stormpath.web.spa.enabled is true, return the SPA view.

>##### POST Requests

>This endpoint accepts a post from the password reset request form, and the only field in this form is the email field. The endpoint should accept the content as application/json or application/x-www-form-urlencoded.

Part solution of #298.

### PR DESCRIPTION
**Pull request by timothyej**
Implements the produces option for `/forgot` according to the framework spec.

#### How to verify

1. Checkout this branch.
2. Clone [express-stormpath-sample-project](https://github.com/stormpath/express-stormpath-sample-project).
3. Use `npm link` and `npm link express-stormpath` to link the library to the example project.
4. Start the server.
5. Navigate to `http://localhost:3000/forgot`.
6. Submit your email.
7. Open up the terminal and execute `$ http POST localhost:3000/forgot email=[your name]@stormpath.com --json`.
8. Verify that you get a `200 OK` response.
9. Modify your configuration and change `web.produces` to `['text/html', '']` (hack because extender currently merges arrays... which needs to be fixed, empty string overwrites second array item.).
10. Verify that step 5-6 works, but that step 7 responds with a `404 Not Found` status.
11. Modify the `web.produces` config to `['application/json', '']`.
12. Verify that step 5 results in a `404 Not Found` and that steps 7-8 work as previous.
13. Modify the `web.produces` config again to `['', '']`.
14. Verify that both steps 5 and 7 result in a ´404 Not Found`.

Fixes #326.

### CODE REVIEW

File: lib/controllers/forgot-password.js [hunk 1]
```diff
@@ -27,45 +27,58 @@ module.exports = function (req, res) {
...
-    error: function (form) {
-      // https://github.com/caolan/forms/issues/96
-      if (req.query.status === 'invalid_sptoken') {
-        return helpers.render(req, res, view, { form: form, status: req.query.status });
+
+    // Handle incoming requests from an browser-like clients (something like
+    // chrome / firefox / etc.).
+    'text/html': function () {
+      if (req.method === 'GET' && config.web.spaRoot) {
+        return res.sendFile(config.web.spaRoot);
```
Comment (robertjd) [inline lib/controllers/forgot-password.js]: This can be removed, as #331 is now handling the serving of the SPA root

**Comment by robertjd** [comment]
+1, looks good!  Please see note about serving the SPA root.

File: lib/controllers/forgot-password.js (hunk 1 above)
Comment (timothyej) [inline lib/controllers/forgot-password.js]: This is now fixed, thank :+1:

File: lib/controllers/forgot-password.js [hunk 2]
```diff
@@ -27,45 +27,54 @@ module.exports = function (req, res) {
...
-      }
 
-      var formErrors = helpers.collectFormErrors(form);
-      helpers.render(req, res, view, { form: form, formErrors: formErrors });
-    },
-    // If we get here, it means the user is doing a simple GET request, so we
-    // should just render the forgot password template.
-    empty: function (form) {
-      helpers.render(req, res, view, { form: form });
+    // Handle incoming requests from an browser-like clients (something like
```
Comment (typerandom) [inline lib/controllers/forgot-password.js]: Remove this comment. It doesn't really add any value. Just bloat.

File: lib/controllers/forgot-password.js [hunk 3]
```diff
@@ -27,45 +27,54 @@ module.exports = function (req, res) {
 
   res.locals.status = req.query.status;
 
-  if (req.method === 'POST' && accepts === 'json') {
-    return application.sendPasswordResetEmail(req.body.email, function (err) {
-      if (err) {
-        logger.info('A user tried to reset their password, but supplied an invalid email address: ' + req.body.email + '.');
+  helpers.handleAcceptRequest(req, res, {
+    // Handle incoming requests from an API-like clients (something like
```
Comment (typerandom) [inline lib/controllers/forgot-password.js]: Remove this comment. It doesn't really add any value. Just bloat.

File: lib/controllers/forgot-password.js [hunk 4]
```diff
@@ -27,45 +27,54 @@ module.exports = function (req, res) {
...
-      helpers.render(req, res, view, { form: form, formErrors: formErrors });
-    },
-    // If we get here, it means the user is doing a simple GET request, so we
-    // should just render the forgot password template.
-    empty: function (form) {
-      helpers.render(req, res, view, { form: form });
+    // Handle incoming requests from an browser-like clients (something like
+    // chrome / firefox / etc.).
+    'text/html': function () {
+      forms.forgotPasswordForm.handle(req, {
```
Comment (typerandom) [inline lib/controllers/forgot-password.js]: Should really all request methods be handled here? I think that you only need to handle GET, maybe POST.

File: lib/controllers/forgot-password.js (hunk 2 above)
Comment (timothyej) [inline lib/controllers/forgot-password.js]: Removed.

File: lib/controllers/forgot-password.js (hunk 3 above)
Comment (timothyej) [inline lib/controllers/forgot-password.js]: Removed.

File: lib/controllers/forgot-password.js (hunk 4 above)
Comment (timothyej) [inline lib/controllers/forgot-password.js]: You're right, it will now only handle `GET` and `POST` requests.

**Comment by typerandom** [comment]
@timothyej This is failing. Are you looking into it?

**Comment by robertjd** [comment]
I rebased `produce-option-change` against `framework-spec-upgrade`, then rebased this branch against `produce-option-change`.  As such, this branch now represents the upgrades to both of those routes.  The tests are interdependent on each other, so this makes the most since.  I added a commit to update the SPA test fixture to use the new configuration format.

Tests are good, merging.

**Comment by timothyej** [comment]
Thanks a lot @robertjd! :)

**Pull request merged**